webapp-django/crashstats/home/static/home/js/home.js

```diff
@@ -0,0 +1,386 @@
+        };
+
+        var url = superSearchApiUrl + '?' + $.param(params, true);
+        $.ajax({
+            url: url,
+            success: onSuperSearchData,
+            error: onQueryError,
+        });
+    }
+
+    /**
+     * Get count of average daily installs per day and per version.
+     */
+    function queryAdi() {
+        var params = {
+            product: product,
+            versions: versions,
+            platforms: platforms,
+            start_date: startDateStr,
+            end_date: endDateStr,
+        };
+
+        var url = adiApiUrl + '?' + $.param(params, true);
+        $.ajax({
+            url: url,
+            success: onAdiData,
+            error: onQueryError,
+        });
+    }
+
+    /**
+     * Get throttle value per build type.
+     */
+    function queryBuildTypes() {
+        var params = {
+            product: product,
+        };
+
+        var url = productBuildTypesApiUrl + '?' + $.param(params, true);
+        $.ajax({
+            url: url,
+            success: onBuildTypesData,
+            error: onQueryError,
+        });
+    }
+
+    /**
+     * Format data received from SuperSearch to make it simpler and closer to
+     * what we will pass to the graph library.
+     */
+    function onSuperSearchData(data) {
+        var i, ln;
+
+        // Make a simple structure to formalize the data, and make it easy to
+        // handle beta versions.
+        var dataStruct = {};
+        for (i = 0, ln = versions.length; i < ln; i++) {
+            dataStruct[versions[i]] = {};
+
+            var currentDay = +startDate;
+            while (currentDay <= +endDate) {
+                var day = dateIsoFormat(new Date(currentDay));
+                dataStruct[versions[i]][day] = 0;
+                currentDay += 24 * 60 * 60 * 1000;
+            }
+        }
+
+        var dates = data.facets.histogram_date;
+        for (i = 0, ln = dates.length; i < ln; i++) {
+            var date = dates[i];
+            var versionsCounts = date.facets.version;
+
+            for (var j = 0, lm = versionsCounts.length; j < lm; j++) {
+                var version = versionsCounts[j].term;
+                var versionIndex = versions.indexOf(version);
+
+                var baseVersion = null;
+                if (versionIndex > -1) {
+                    baseVersion = version;
+                }
+                else if (versionIndex === -1) {
+                    // If this version was not found in the requested list of
+                    // versions, then it must be a beta version.
+                    for (var k = betaVersions.length - 1; k >= 0; k--) {
+                        if (version.indexOf(betaVersions[k]) > -1) {
+                            baseVersion = betaVersions[k];
+                        }
+                    }
+                }
+
+                dataStruct[baseVersion][date.term.substring(0, 10)] += versionsCounts[j].count;
+            }
+        }
```

Comment:
I read it but my head's spinning. What does this whole thing do? I know it's about filling up `dataStruct` but some human-readable summary would be nice.